hack/preview.sh

```diff
@@ -86,6 +86,9 @@ fi
 [ -n "${BUILD_SERVICE_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/build-service\")) |=.newName=\"${BUILD_SERVICE_IMAGE_REPO}\"" $ROOT/components/build/build-service/kustomization.yaml
 [ -n "${BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/build-service\")) |=.newTag=\"${BUILD_SERVICE_IMAGE_TAG}\"" $ROOT/components/build/build-service/kustomization.yaml
 [[ -n "${BUILD_SERVICE_PR_OWNER}" && "${BUILD_SERVICE_PR_SHA}" ]] && yq -i e "(.resources[] | select(. ==\"*github.com/redhat-appstudio/build-service*\")) |= \"https://github.com/${BUILD_SERVICE_PR_OWNER}/build-service/config/default?ref=${BUILD_SERVICE_PR_SHA}\"" $ROOT/components/build/build-service/kustomization.yaml
+[ -n "${JVM_BUILD_SERVICE_IMAGE_REPO}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/hacbs-jvm-controller\")) |=.newName=\"${JVM_BUILD_SERVICE_IMAGE_REPO}\"" $ROOT/components/build/jvm-build-service/kustomization.yaml
+[ -n "${JVM_BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/hacbs-jvm-controller\")) |=.newTag=\"${JVM_BUILD_SERVICE_IMAGE_TAG}\"" $ROOT/components/build/jvm-build-service/kustomization.yaml
```

Comment:
The name is `hacbs-jvm-operator`

Reply:
Is it? 

I see https://quay.io/repository/redhat-appstudio/hacbs-jvm-controller

Here are the 5 JVM images I see when I filter on "jvm":

<span ng-transclude="">

R                                  redhat-appstudio             hacbs-jvm-build-request-processor | Yesterday at 5:45 PM |   |  
-- | -- | -- | --
R                                  redhat-appstudio             hacbs-jvm-cache | Yesterday at 5:48 PM |   |  
R                                  redhat-appstudio             hacbs-jvm-controller | Yesterday at 5:46 PM |   |  
R                                  redhat-appstudio             hacbs-jvm-sidecar | Yesterday at 5:29 PM |   |  
R                                  redhat-appstudio             hacbs-jvm-dependency-analyser | Yesterday at 5:46 PM |   |  

</span>R[ redhat-appstudio hacbs-jvm-build-request-processor ](https://quay.io/repository/redhat-appstudio/hacbs-jvm-build-request-processor)	Yesterday at 5:45 PM 		
R[ redhat-appstudio hacbs-jvm-cache ](https://quay.io/repository/redhat-appstudio/hacbs-jvm-cache)	Yesterday at 5:48 PM 		
R[ redhat-appstudio hacbs-jvm-controller ](https://quay.io/repository/redhat-appstudio/hacbs-jvm-controller)	Yesterday at 5:46 PM 		
R[ redhat-appstudio hacbs-jvm-sidecar ](https://quay.io/repository/redhat-appstudio/hacbs-jvm-sidecar)	Yesterday at 5:29 PM 		
R[ redhat-appstudio hacbs-jvm-dependency-analyser ](https://quay.io/repository/redhat-appstudio/hacbs-jvm-dependency-analyser)	Yesterday at 5:46 PM

Reply:
oh wait ... it is substituting on https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/jvm-build-service/kustomization.yaml#L22 isn't it

will update

Reply:
no ... the select is 

`select(.name==\"quay.io/redhat-appstudio/hacbs-jvm-controller\"))`

so isn't it on the image name @Michkov ?

Reply:
yes, it's selecting from images the dict which has name 'quay.io/redhat-appstudio/hacbs-jvm-controller' but name is hacbs-jvm-cache -> https://github.com/redhat-appstudio/infra-deployments/blob/main/components/build/jvm-build-service/kustomization.yaml#L16

Reply:
no with this change I am not updating for hacbs-jvm-cache

right now the e2e integration here with jvm-bulid-service and openshift/release is only for the golang controller/operator image

I'm am not doing the jvm based images yet.  I've have to sort our some java/maven dependencies wrt creating builder image for use in openshift/release, and then we can follow up with overrides of hacbs-jvm-cache, sidecar, etc.

Reply:
oh ... you mean the index into the images array, don't you

OK *now* I follow (I think)

Reply:
though I thought that `yq` call cycled through all of the images array, so I'm still confused

Reply:
```
$ export JVM_BUILD_SERVICE_IMAGE_TAG=abc
$ [ -n "${JVM_BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"quay.io/redhat-appstudio/hacbs-jvm-controller\")) |=.newTag=\"${JVM_BUILD_SERVICE_IMAGE_TAG}\"" components/build/jvm-build-service/kustomization.yaml
$ grep abc components/build/jvm-build-service/kustomization.yaml
$ [ -n "${JVM_BUILD_SERVICE_IMAGE_TAG}" ] && yq -i e "(.images.[] | select(.name==\"hacbs-jvm-operator\")) |=.newTag=\"${JVM_BUILD_SERVICE_IMAGE_TAG}\"" components/build/jvm-build-service/kustomization.yaml
$ grep abc components/build/jvm-build-service/kustomization.yaml
    newTag: abc
```

Reply:
yep that illustrates it ;-)

I also see what was confusing me .... in the build-service one, the image name is used for both name and newName

will update shortly thanks

Reply:
updated @Michkov thanks